ifrs17/Utils/Queries.ipynb

```diff
@@ -183,18 +183,18 @@
         {
             "cell_type": "code",
             "source": [
-                "public static async Task<Dictionary<string, YieldCurve>> LoadLockedInYieldCurveAsync(this IQuerySource querySource, Args args, ",
-                "\n                                                                                                       IEnumerable<DataNodeData> dataNodes)",
+                "public static async Task<Dictionary<string, YieldCurve>> LoadLockedInYieldCurveAsync(this IQuerySource querySource, Args args, IEnumerable<DataNodeData> dataNodes)",
+                "\n",
                 "\n{",
                 "\n    var lockedInYieldCurveByGoc = new Dictionary<string, YieldCurve>();",
                 "\n    foreach (var dn in dataNodes.Where(x => x.ValuationApproach == ValuationApproaches.BBA))",
                 "\n    {",
                 "\n        var argsNew = args with {Year = dn.Year, Month = dn.Month, Scenario = args.Scenario};",
-                "\n        var loadedYc = (await querySource.LoadCurrentParameterAsync<YieldCurve>(argsNew, x => x.Currency, x => x.Currency == dn.ContractualCurrency));",
-                "\n        ",
-                "\n        if(!loadedYc.TryGetValue(dn.ContractualCurrency, out var lockedYc))",
+                "\n        var loadedYc = (await querySource.LoadCurrentParameterAsync<YieldCurve>(argsNew, x => x.Currency, x => x.Currency == dn.ContractualCurrency && x.Name == dn.YieldCurveName));",
+                "\n                                    ",
+                "\n        if (!loadedYc.TryGetValue(dn.ContractualCurrency, out var lockedYc) || lockedYc == null)",
```

Comment:
How can this happen? If on the dictionary, the bool is false anyway. Otherwise, lockedYc should never be null.

Reply:
@dtrzesniak Daniel, do you mean checking additional "|| lockedYc == null" ? That one makes no sense since TryGetValue controls enough.

Reply:
Exactly... and there should never be a null value in this dictionary

Reply:
@dtrzesniak Fixed with the last commit.